services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_insights/history_and_recommendations/index.tsx

```diff
@@ -74,14 +74,14 @@ const HistoryAndRecommendations = ({ teacherClassroomIds, classroom, needsEmailV
   }, [location.pathname, navigate]);
 
   React.useEffect(() => {
-    if (byClass || selectedStudentId) { return }
+    if (byClass || selectedStudentId || !studentHistories) { return }
 
     const match = location.pathname.match(/by_student\/(\d+)$/);
     const studentId = match ? Number(match[1]) : null;
 
     if (studentId && studentHistories?.find(s => s.user_id === studentId)) {
```

Comment:
Not necessary, but given that we are now guaranteeing that `studentHistories` exists in the early return above, we could remove the `?.` safe accessor from this line as well as the one below.